### PR TITLE
Fix rewind data index

### PR DESCRIPTION
**Pull request by dangershony**
Rewind Data Index was indexing too much (new UTXO as well as spent UTXO) and it was overwriting UTXO that where not spend but still pulled from disk (this is due to the nature of how coinview works where all the UTXOs of a trx are loaded even though not all inputs are spent n the block)

### CODE REVIEW

**Comment by quantumagi** [comment]
@dangershony , does this also fix the `ConsensusManager_Fork_Occurs_When_Stake_Coins_Are_Spent_And_Found_In_Rewind_Data` test case? I believe this was the cause for the test to fail.

File: src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs [hunk 1]
```diff
@@ -193,6 +193,9 @@ protected virtual void InitializeTransactionBuilder(TransactionBuildContext cont
             this.AddSecrets(context);
             this.FindChangeAddress(context);
             this.AddFee(context);
+
+            if (context.Time.HasValue)
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs]: Ah, so we didn't add this in here when it got added to the federation wallet?

File: src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs [hunk 2]
```diff
@@ -421,6 +421,19 @@ public void SaveChanges(IList<UnspentOutputs> unspentOutputs, IEnumerable<TxOut[
                         cacheItem.UnspentOutputs.Spend(unspent);
 
                         this.logger.LogTrace("Cache item after spend {0}:'{1}'.", nameof(cacheItem.UnspentOutputs), cacheItem.UnspentOutputs);
+
+                        if (this.rewindDataIndexCache != null)
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs]: If this gets moved here, where do new outpoints get added into the `rewindDataIndexCache`?

Comment (dangershony) [reply src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs]: Rewind index only points to utxo that are spent and currently reside in the rewind store so we don't need to index new outputs

**Comment by dangershony** [comment]
Synced stratis testnet this is now good to push

**Pull request merged**